Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Github5623.xaml.cs

```diff
@@ -0,0 +1,201 @@
+using Xamarin.Forms.Xaml;
+using System;
+using System.Security.Cryptography;
+
+#if UITEST
+using Xamarin.UITest;
+using NUnit.Framework;
+using Xamarin.Forms.Core.UITests;
+using System.Linq;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+#if UITEST
+	[Category(UITestCategories.CollectionView)]
+#endif
+#if APP
+	[XamlCompilation(XamlCompilationOptions.Compile)]
+#endif
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 5623, "CollectionView with Incremental Collection (RemainingItemsThreshold)", PlatformAffected.All)]
+	public partial class Github5623 : TestContentPage
+	{
+		int _itemCount = 10;
+		int _maximumItemCount = 100;
+		int _pageSize = 10;
+		static SemaphoreSlim SemaphoreSlim = new SemaphoreSlim(1, 1);
+
+		public Github5623()
+		{
+#if APP
+			Device.SetFlags(new List<string> { CollectionView.CollectionViewExperimental });
+
+			InitializeComponent();
+
+			BindingContext = new ViewModel5623();
+#endif
+		}
+
+		protected override void Init()
+		{
+
+		}
+
+		async void CollectionView_RemainingItemsThresholdReached(object sender, System.EventArgs e)
+		{
+			await SemaphoreSlim.WaitAsync();
+			try
+			{
+				var itemsSource = (sender as CollectionView).ItemsSource as ObservableCollection<Model5623>;
+				var nextSet = await GetNextSetAsync();
+
+				// nothing to add
+				if (nextSet.Count == 0)
+					return;
+
+				Device.BeginInvokeOnMainThread(() =>
+				{
+					foreach (var item in nextSet)
+					{
+						itemsSource.Add(item);
+					}
+				});
+
+				System.Diagnostics.Debug.WriteLine("Count: " + itemsSource.Count);
+			}
+			finally
+			{
+				SemaphoreSlim.Release();
+			}
+		}
+
+		void CollectionView_OnScrolled(object sender, ItemsViewScrolledEventArgs e)
+		{
+			Label1.Text = "HorizontalDelta: " + e.HorizontalDelta;
+			Label2.Text = "VerticalDelta: " + e.VerticalDelta;
+			Label3.Text = "HorizontalOffset: " + e.HorizontalOffset;
+			Label4.Text = "VerticalOffset: " + e.VerticalOffset;
+			Label5.Text = "FirstVisibleItemIndex: " + e.FirstVisibleItemIndex;
+			Label6.Text = "CenterItemIndex: " + e.CenterItemIndex;
+			Label7.Text = "LastVisibleItemIndex: " + e.LastVisibleItemIndex;
+		}
+
+		async Task<ObservableCollection<Model5623>> GetNextSetAsync()
+		{
+			return await Task.Run(() =>
+			{
+				var collection = new ObservableCollection<Model5623>();
+				var count = _pageSize;
+
+				if (_itemCount + count > _maximumItemCount)
+					count = _maximumItemCount - _itemCount;
+
+				for (var i = _itemCount; i < _itemCount + count; i++)
+				{
+					collection.Add(new Model5623((BindingContext as ViewModel5623).ItemSizingStrategy == ItemSizingStrategy.MeasureAllItems)
+					{
+						Text = i.ToString(),
+						BackgroundColor = i % 2 == 0 ? Color.AntiqueWhite : Color.Lavender
+					});
+				}
+
+				_itemCount += count;
+
+				return collection;
+			});
+		}
+
+#if UITEST
+		[Test]
+		public void CollectionViewInfiniteScroll()
+		{
+			RunningApp.WaitForElement (q => q.Marked ("CollectionView"));
```

Comment:
Locally this times out (on Android). No clue why, adding an explicit `AutomationId` to the `CollectionView` seems to help. Personally I like to avoid the object names as identifiers (i.e. `x:Name="CollectionView"`) but not sure if that has anything to do with it

Reply:
Appended `5623` to the name. Let's see if it fixes the issue. Looking up objects by their name should in theory work without the need for an explicit automation id.

Reply:
I agree! That should definitely work. I added a `Repl` call in there and this query came up empty. Since the `AutomationId` is meant to be used for this purpose I tried that and it worked. I didn't try this variation though :)